astropy/units/format/vounit.py

```diff
@@ -143,18 +152,7 @@ def _get_unit_name(cls, unit):
                 )
 
         name = super()._get_unit_name(unit)
-
-        if unit in cls._custom_units.values():
```

Comment:
I think this part is unique to `VOUnit`, so should be kept.

Aside: very worried if there are no test failures with its removal!

Reply:
This check has been moved to `VOUnit._validate_unit()`, you just have to keep in mind that in `_get_unit_name()` `unit` is a `UnitBase`, but in `_validate_unit()` it is a `str`, so we have check `_custom_units` keys instead of values.

The intended behavior is being tested: https://github.com/astropy/astropy/blob/3be7d5631a03c35811f0cfd16b3c41f670aa7724/astropy/units/tests/test_format.py#L756-L768